watertap/property_models/NaCl_prop_pack.py

```diff
@@ -741,49 +741,34 @@ def calculate_scaling_factors(self):
             iscale.set_scaling_factor(self.enth_flow,
                                       iscale.get_scaling_factor(self.flow_mass_phase_comp['Liq', 'H2O'])
                                       * iscale.get_scaling_factor(self.enth_mass_phase['Liq']))
-
         # transforming constraints
-        # property relationships with no index, simple constraint
-        v_str_lst_simple = ('osm_coeff', 'pressure_osm')
-        for v_str in v_str_lst_simple:
-            if self.is_property_constructed(v_str):
-                v = getattr(self, v_str)
-                sf = iscale.get_scaling_factor(v, default=1, warning=True)
-                c = getattr(self, 'eq_' + v_str)
-                iscale.constraint_scaling_transform(c, sf)
+        if self.is_property_constructed('pressure_osm'):
+            sf = iscale.get_scaling_factor(self.pressure_osm, default=1, warning=True)
+            iscale.constraint_scaling_transform(self.eq_pressure_osm, sf)
+        if self.is_property_constructed('osm_coeff'):
+            sf = iscale.get_scaling_factor(self.mass_frac_phase_comp['Liq', 'NaCl'], default=1, warning=True)
+            iscale.constraint_scaling_transform(self.eq_osm_coeff, sf**1.25)
 
         # property relationships with phase index, but simple constraint
-        for v_str in ('flow_vol_phase', 'visc_d_phase', 'diffus_phase'):
+        for v_str in ('visc_d_phase', 'enth_mass_phase', 'flow_vol_phase', 'diffus_phase'):
             if self.is_property_constructed(v_str):
-                v = getattr(self, v_str)
-                sf = iscale.get_scaling_factor(v['Liq'], default=1, warning=True)
-                c = getattr(self, 'eq_' + v_str)
-                iscale.constraint_scaling_transform(c, sf)
+                sf = iscale.get_scaling_factor(self.component(v_str)['Liq'], default=1, warning=True)
+                iscale.constraint_scaling_transform(self.component('eq_'+v_str), sf)
 
-        for v_str in ('dens_mass_phase', 'enth_mass_phase'):
-            if self.is_property_constructed(v_str):
-                v = getattr(self, v_str)
-                sf = iscale.get_scaling_factor(v['Liq'], default=1, warning=True)
-                c = getattr(self, 'eq_' + v_str)
-                iscale.constraint_scaling_transform(c, sf*10.)
+        if self.is_property_constructed('dens_mass_phase'):
+            sf = iscale.get_scaling_factor(self.mass_frac_phase_comp['Liq', 'NaCl'])
+            iscale.constraint_scaling_transform(self.eq_dens_mass_phase, sf/self.params.dens_mass_param['1'])
```

Comment:
This looks odd to me. I'm not sure why the density is tied to the scaling factor for the NaCl mass fraction. It should just be `1/self.params.dens_mass_param['0']` (note the parameter is indexed by 0 rather than 1). 

Oh wait its easier than that, we have a default scaling factor being set for the dens_mass_phase variable. So the scaling factor can be:
sf = iscale.get_scaling_factor(self.dens_mass_phase['Liq'])